### PR TITLE
Feature/add id requirements

### LINKED ISSUE #1729: Store if a given election requires ID, and if so what the requirements are
Some UK elections require ID in order to vote in person.

Before 2023 all of these elections were in Northern Ireland. This meant the only logic needed was to check the country the election was being held in and show text accordingly.

After 2023 the situation gets tricky.

All English elections will require ID (but this is phasing in, with general election law being activated later this year, so a parliamentary by-election now wouldn't require ID).

Welsh-run elections aren't going to need ID, but PCC and parliamentary elections in Wales are. Same for Scotland.

This means that EE should store, per ballot, if the election is going to require ID or not. 

The ID requirements might change over time or per ballot as well, so we should find a way to list the forms of ID (or at least a link to a page /object that knows the requirements).

### PR DESCRIPTION
**Pull request by Bekabyx**
This work aims to add a field, `requires_voter_id` to the elections database, populate the existing records and ensure that any new elections created via the ID creator have their ID requirements assigned correctly. This is the preliminary work for doing the ID messaging ([Trello](https://trello.com/c/UvMnJkuS/3207-voter-id-messaging)).  Closes #1729 

Also, added a nice dropdown list to the admin panel where ID requirements can be manually assigned/overridden.
Screenshot 2023-03-09 at 16 04 10

Due to the election ID relying on divisions having territory codes, this PR is blocked by #1865 for the moment.

To test this locally:
- Make sure you've done a fresh requirements install
- Run the migration
- Check your elections table for anything that you know should need ID (I'd suggest a local election in ENG/WAL/SCT in May 2023) and check that the `requires_voter_id` column is set to `EA-2022` (n.b. it will be `EFA-2002` if you're checking a NI election)
- Run EE locally and use the ID creator to make any GB parliamentary election in November for any constituency you like
- Find the new election in your elections table and verify that it has `requires_voter_id` set to `EA-2022`
- Log in to your local admin panel, find the election you created, and set the voter id requirements to something else using the dropdown 
- Save it and check it's updated in your db.

```[tasklist]
### PR Checklist

- [x] References a specific issue or if not, describes the bug or feature in detail
- [x] Note what card in Trello this work relates to
- [x] Instructions for how reviewers can test the code locally
- [x] Tests have been added and/or updated
- [x] Screenshot of the feature/bug fix (if applicable)
- [x] No unintentional console.logs left behind after debugging
- [x] Did I use the clear and concise names for variables and functions?
- [x] Added any comments to make new functions clearer
- [x] Did I add or updated any dependencies? Explain why.
- [x] Have I rebased with the latest version of master?
```

### CODE REVIEW

File: every_election/apps/elections/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def test_get_voter_id_requirement_no_division(self):
+        election = ElectionFactory(election_id=self.default_election_id)
+        election.division = None
+        assert get_voter_id_requirement(election) is None
+
+    def test_get_voter_id_requirement_fake_territory(self):
+        election = ElectionFactory(election_id=self.default_election_id)
+        with pytest.raises(ValueError):
+            election.division.territory_code = "BBQ"
```
Comment (symroe) [inline every_election/apps/elections/tests/test_utils.py:29]: :hamburger:

**Review by symroe — APPROVED** [review]
Nice one!

File: every_election/apps/elections/tests/test_utils.py (hunk 1 above)
Comment (Bekabyx) [reply every_election/apps/elections/tests/test_utils.py:29]: I was hungry 😆

**Comment by VirginiaDooley** [comment]
Confirmed with `every_election=# SELECT * FROM elections_election WHERE election_id = 'local.antrim-and-newtownabbey.three-mile-water.2023-05-18';

**Review by VirginiaDooley — APPROVED** [review]
Tested locally, looks good! 

Confirmed with `every_election=# SELECT * FROM elections_election WHERE election_id = 'local.antrim-and-newtownabbey.three-mile-water.2023-05-18';

Screenshot 2023-03-13 at 11 54 15 AM

**Pull request merged**